examples/c/ex_col_store.c

```diff
@@ -0,0 +1,535 @@
+    uint8_t temp;
+
+    /* Open cursors needed by the join. */
+    error_check(
+      session->open_cursor(session, "join:table:weather(hour,temp)", NULL, NULL, &join_cursor));
+    error_check(
+      session->open_cursor(session, "index:weather:hour", NULL, NULL, &start_time_cursor));
+    error_check(session->open_cursor(session, "index:weather:hour", NULL, NULL, &end_time_cursor));
+
+    /*
+     * Select values WHERE (hour >= start AND hour <= end). Find the starting record closest to
+     * desired start time.
+     */
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    if (exact == -1) {
+        ret = start_time_cursor->next(start_time_cursor);
+        if (ret == WT_NOTFOUND)
+            return ret;
+        else
+            error_check(ret);
+    }
+
+    error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    if (exact == 1) {
+        ret = end_time_cursor->prev(end_time_cursor);
+        if (ret == WT_NOTFOUND)
+            return ret;
+        else
+            error_check(ret);
+    }
+
+    error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+
+    /* Initialize minimum temperature and maximum temperature to temperature of the first record. */
+    ret = join_cursor->next(join_cursor);
+    if (ret == WT_NOTFOUND)
+        return ret;
+    else
+        error_check(ret);
+
+    error_check(join_cursor->get_key(join_cursor, &recno));
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *min_temp = temp;
+    *max_temp = temp;
+
+    /* Iterating through found records between start and end time to find the min & max temps. */
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+
+        *min_temp = WT_MIN(*min_temp, temp);
+        *max_temp = WT_MAX(*max_temp, temp);
+    }
+
+    /*
+     * If WT_NOTFOUND is hit at this point, it is because we have traversed through all temperature
+     * records, hence we return 0 to the calling function to signal success. Otherwise an internal
+     * error was hit.
+     */
+    if (ret != WT_NOTFOUND)
+        error_check(ret);
+
+    return (0);
+}
+
+/*
+ * average_data --
+ *     Obtains the average data across all fields given a specific location.
+ */
+void
+average_data(WT_SESSION *session)
+{
+    WT_CURSOR *loc_cursor;
+    unsigned int count;
+    uint16_t hour, loc_lat, loc_long, pressure;
+    uint8_t feels_like_temp, humidity, temp, wind;
+    const char *country, *day;
+    WT_DECL_RET;
+    /* rec_arr holds the sum of the records in order to obtain the averages. */
+    unsigned int rec_arr[NUM_REC] = {0, 0, 0, 0, 0};
```

Comment:
The variable declaration order should be as follows.

```suggestion
    WT_CURSOR *loc_cursor;
    WT_DECL_RET;
    unsigned int count;
    /* rec_arr holds the sum of the records in order to obtain the averages. */
    unsigned int rec_arr[NUM_REC];
    uint16_t hour, loc_lat, loc_long, pressure;
    uint8_t feels_like_temp, humidity, temp, wind;
    const char *country, *day;
```